src/unwinder/arch/aarch64.rs

```diff
@@ -52,13 +55,47 @@ impl ops::IndexMut<gimli::Register> for Context {
         match reg {
             Register(0..=30) => &mut self.gp[reg.0 as usize],
             AArch64::SP => &mut self.sp,
+            #[cfg(not(target_abi = "softfloat"))]
             Register(64..=95) => &mut self.fp[(reg.0 - 64) as usize],
             _ => unimplemented!(),
         }
     }
 }
 
 #[naked]
+#[cfg(target_abi = "softfloat")]
+pub extern "C-unwind" fn save_context(f: extern "C" fn(&mut Context, *mut ()), ptr: *mut ()) {
+    // No need to save caller-saved registers here.
```

Comment:
Could you extract the asm using macro to avoid duplication? See riscv for example

Reply:
Yeah sure, will do.

Reply:
Please take another look of riscv code to avoid code duplication